app/assets/javascripts/basetypes/DataElement.js

```diff
@@ -1,36 +1,69 @@
 const mongoose = require('mongoose');
 const Code = require('./Code.js');
+const changeCase = require('change-case');
 
 const [Schema] = [mongoose.Schema];
 
-function extendSchema(TSchema, definition, options) {
-  return new Schema(Object.assign({}, TSchema.obj, definition), options);
-}
+function DataElementSchema(add, options) {
+  const extended = new Schema({
+    data_element_codes: { type: [Code] },
+    description: { type: String },
+  }, options);
+
+  if (add) {
+    extended.add(add);
+  }
+
+  extended.methods.get = function get(path, type) {
+    let adhoc;
+    const pathSnake = changeCase.snakeCase(path);
+    if (type) {
+      adhoc = Schema.interpretAsType(path, type, this.schema.options) || Schema.interpretAsType(pathSnake, type, this.schema.options);
```

Comment:
Please add some comments for this get function. Seems a like a big change from the previous version of "get()"